src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/OutputFormatterSelector.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Formatters;
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    public abstract class OutputFormatterSelector
```

Comment:
What's the benefit of an abstract class instead of an interface?

Reply:
It has a better set of tradeoffs for something like this (IMO).

With an abstract class we can easily add new functionality or overloads without breaking anything. When something is implementation-ey, the requirements frequently change. We'd do something like add a new API, obsolete the old one, and then remove it in the next major release. You can't do that with an interface, you have to make a new interface.

Interfaces are better when the contract is minimal, the requirements seldom change, or when we expect lots of implementations. 

---- 

I'm open to feedback if you think this is the wrong direction

Reply:
I don't have a clear preference in this case. An abstract class with only a single abstract method seems a bit strange, but I understand that the class is better than the interface for future changeability.

> Interfaces are better when the contract is minimal, the requirements seldom change

I kind of expect a content negotiation interface to be minimal and rarely changing. Ideally, we could have something like this:
```
public interface IContentNegotiator
{
  IOutputFormatter SelectFormatter(IEnumerable<IOutputFormatter> formatters, MediaTypeCollection mediaTypes);
}
```
But then the selection logic would not take into account if the result object can be serialized by the selected output formatter and I guess that's important too. :-)

BTW I got to this PR, because I wanted to do content negotiation in an exception handling middleware. Currently the easiest way of doing it is calling the `SelectFormatter` by reflection. This PR would allow me to do that without any reflection hack, so as far as I'm concerned, this is perfect.